telegramium-high/src/main/scala/telegramium/bots/high/package.scala

```diff
@@ -0,0 +1,144 @@
+package telegramium.bots
+
+package object high {
+
+  type KeyboardButton = telegramium.bots.KeyboardButton
+  object KeyboardButton {
+    /**
+     * Text will be sent to the bot as a message when the button is pressed.
+     */
+    def text(text: String): KeyboardButton = new KeyboardButton(text)
+
+    /**
+     * The user's phone number will be sent as a contact when the button is pressed.
+     */
+    def requestContact(text: String): KeyboardButton = new KeyboardButton(text, requestContact = Some(true))
+
+    /**
+     * The user's current location will be sent when the button is pressed
+     */
+    def requestLocation(text: String): KeyboardButton = new KeyboardButton(text, requestLocation = Some(true))
+
+    /**
+     * The user will be asked to create a poll and send it to the bot when the button is pressed
+     */
+    def requestPoll(text: String, `type`: KeyboardButtonPollType) = new KeyboardButton(text, requestPoll = Some(`type`))
+  }
+
+  type ReplyKeyboardMarkup = telegramium.bots.ReplyKeyboardMarkup
+  object ReplyKeyboardMarkup {
+    /**
+     * Creates a reply keyboard markup with one button
+     */
+    def singleButton(
+      button: KeyboardButton,
+      resizeKeyboard: Option[Boolean] = Option.empty,
+      oneTimeKeyboard: Option[Boolean] = Option.empty,
+      selective: Option[Boolean] = Option.empty
+    ) = new ReplyKeyboardMarkup(
+      keyboard = List(List(button)),
+      resizeKeyboard = resizeKeyboard,
+      oneTimeKeyboard = oneTimeKeyboard,
+      selective = selective
+    )
+
+    /**
+     * Creates a reply keyboard markup with multiple buttons on a single row
+     */
+    def singleRow(
+      row: List[KeyboardButton],
+      resizeKeyboard: Option[Boolean] = Option.empty,
+      oneTimeKeyboard: Option[Boolean] = Option.empty,
+      selective: Option[Boolean] = Option.empty
+    ) = new ReplyKeyboardMarkup(
+      keyboard = List(row),
+      resizeKeyboard = resizeKeyboard,
+      oneTimeKeyboard = oneTimeKeyboard,
+      selective = selective
+    )
+
+    /**
+     * Creates a reply keyboard markup with multiple buttons on a single column
+     */
+    def singleColumn(
+      column: List[KeyboardButton],
+      resizeKeyboard: Option[Boolean] = Option.empty,
+      oneTimeKeyboard: Option[Boolean] = Option.empty,
+      selective: Option[Boolean] = Option.empty
+    ) = new ReplyKeyboardMarkup(
+      keyboard = column.map(List(_)),
+      resizeKeyboard = resizeKeyboard,
+      oneTimeKeyboard = oneTimeKeyboard,
+      selective = selective
+    )
+  }
+
+  type InlineKeyboardButton = telegramium.bots.InlineKeyboardButton
+  object InlineKeyboardButton {
+    /**
+     * Creates an inline keyboard button that sends a callback query to bot when pressed
+     */
+    def callbackData(text: String, callbackData: String): InlineKeyboardButton =
```

Comment:
@johnspade why do we need these methods? I mean just calling case class constructor looks shorter.. Can you elaborate on your idea

Reply:
They're discoverable and prevent you from passing wrong number of arguments to case class constructors

Reply:
Good point, let's have it.